src/QuadVelocityController.cpp

```diff
@@ -303,10 +309,20 @@ bool QuadVelocityController::getBatteryAtTime(
     // Wait until there's a message with stamp >= time
     if (!waitForBatteryAtTime(time, timeout))
```

Comment:
The problem of the battery messages limiting the update rate of the total controller is still here. This function will wait for 0.1 seconds (the update_timeout) parameter. I think this needs to wait for a battery at time (time - battery_timeout).